src/providers/swift.rs

```diff
@@ -0,0 +1,71 @@
+use super::Provider;
+use crate::nixpacks::{
+    app::App,
+    environment::{Environment, EnvironmentVariables},
+    nix::pkg::Pkg,
+    phase::{BuildPhase, SetupPhase, StartPhase},
+};
+use anyhow::Result;
+
+pub struct SwiftProvider {}
+
+impl Provider for SwiftProvider {
+    fn name(&self) -> &str {
+        "swift"
+    }
+
+    fn detect(&self, app: &App, _env: &Environment) -> Result<bool> {
+        Ok(app.includes_file("Package.swift")
+            || (app.includes_file("Package.swift") && app.includes_file("Package.resolved")))
+    }
+
+    fn setup(&self, _app: &App, _env: &Environment) -> Result<Option<SetupPhase>> {
+        let pkg = Pkg::new("swift");
+
+        Ok(Some(SetupPhase::new(vec![pkg])))
+    }
+
+    fn build(&self, _app: &App, _env: &Environment) -> Result<Option<BuildPhase>> {
+        Ok(Some(BuildPhase::new(
+            "swift build -c release --static-swift-stdlib".to_owned(),
+        )))
+    }
+
+    fn start(&self, app: &App, _env: &Environment) -> Result<Option<StartPhase>> {
+        let raw_paths = app.find_files("Sources/**/main.swift")?;
```

Comment:
Is the directory Sources configurable?

Reply:
I think not?
https://forums.swift.org/t/customizing-location-of-source-directory/725/4